src/ng/location.js

```diff
@@ -48,17 +48,20 @@ function parseAppUrl(relativeUrl, locationObj) {
   }
 }
 
+function startsWith(haystack, needle) {
```

Comment:
One refactoring at a time :-)
At this stage this would only be used here and in $sanitize. So I felt it was simpler right now not to move it out.
But sure, it would be fine to do so.